### PR TITLE
fix(ext/node): allow automatic worker_thread termination

### PR DESCRIPTION
**Pull request by satyarohith**
This PR fixes node_worker to auto exit after script execution is completed.  It doesn't update Deno worker behaviour as the spec expects users to manually terminate the web worker.

Closes https://github.com/denoland/deno/issues/22629

### CODE REVIEW

**Comment by mash-graz** [comment]
Thanks for working on this issue!

But just a small remark concerning the questionable standard compliant behavior of `Deno.Worker`.

Even if you add this code fragment:
```
w.onclose = () => { console.log("worker closed"); w.terminate() };
```
the application will not stop!

It will get the `close`-event, but the additional `w.terminate()` doesn't change the termination issue affecting the main process. :(

**Comment by mash-graz** [comment]
This whole issue looks close related to #14318

File: runtime/js/99_main.js [hunk 1]
```diff
@@ -280,50 +280,59 @@ function postMessage(message, transferOrOptions = {}) {
...
+        globalThis.dispatchEvent,
+        globalThis,
+      );
+    }
+    while (!isClosing) {
+      const op = op_worker_recv_message();
+      // In a Node.js worker, unref() the op promise to prevent it from
+      // keeping the event loop alive. This avoids the need to explicitly
+      // call self.close() or worker.terminate().
+      if (workerName === "$DENO_STD_NODE_WORKER_THREAD") {
```
Comment (mmastrac) [inline runtime/js/99_main.js:296]: I don't think this fix will be very robust long-term. Instead of testing this here, we should initialize a worker with a "close_on_idle" flag.

File: runtime/web_worker.rs [hunk 2]
```diff
@@ -757,6 +757,10 @@ impl WebWorker {
           return Poll::Ready(Err(e));
         }
 
+        if self.name == "$DENO_STD_NODE_WORKER_THREAD" {
```
Comment (mmastrac) [inline runtime/web_worker.rs:760]: Ditto here -- let's use a flag.

File: runtime/js/11_workers.js [hunk 3]
```diff
@@ -119,7 +114,7 @@ class Worker extends EventTarget {
       }
     }
 
-    this.#name = name;
+    this.#name = name || "";
```
Comment (bartlomieju) [inline runtime/js/11_workers.js:117]: Why this change?

Comment (satyarohith) [reply runtime/js/11_workers.js:117]: There are some unrelated changes that got included during merge. I'll revert them.

**Comment by satyarohith** [comment]
FYI: https://github.com/denoland/deno/pull/22826 is a pre-requisite for this PR. cc @bartlomieju

**Review by bartlomieju — APPROVED** [review]
LGTM, I confirmed the test hangs on main, but passes in Node.js and this PR. Nice work
